packages/wonder-blocks-form/src/components/text-area.tsx

```diff
@@ -273,10 +273,10 @@ const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
                     className={className}
                     style={[...getStyles(), style]}
                     value={value}
-                    onChange={handleChange}
+                    onChange={disabled ? undefined : handleChange}
```

Comment:
**question:** What would happen if the `TextArea` instance is a controlled one? should we still trigger `onChange` in case the parent component updates its `value` prop?

Reply:
Good question! If I'm understanding the question correctly, the parent component will still be able to update the `value` prop and have it update what is shown in the textarea:

https://github.com/user-attachments/assets/7dd79f1d-db51-4ed6-b05d-e97f7cfe2e03

So if something other than the TextArea was controlling the value, it will still update (just the TextArea component won't call the onChange callback)

Kinda related - when I set `onChange` to undefined when it is disabled though, I also set the `readOnly` prop to `true` because of this warning (see below). `readOnly` also prevents the textarea cursor from showing when it is disabled :)

<img width="539" alt="Screenshot 2024-08-02 at 9 41 22 AM" src="https://github.com/user-attachments/assets/da4eb1e2-f7e9-4287-80e8-fdc6b29792a0">

Let me know if you have thoughts around this or if I'm missing something!

Reply:
Thanks for the context! My reasoning about enabling `onChange` in those cases is that when the component is in controlled mode, then that means that both `value` and `onChange` are configured ([docs](https://khan.github.io/wonder-blocks/?path=/docs/packages-form-textarea--docs#controlled)). I totally understand your point here, but I feel that allowing value/onChange to work in expected ways would be better as folks will know that the API will remain consistent.

Maybe you could only modify that behavior for when it is disabled and it is controlled?

Reply:
Gotcha! Turns out if I change this back, the `onChange` handler doesn't get called anyways since the TextArea is in readonly mode also if it is disabled (so we didn't need this extra logic 😄 ). This simplifies things and is more consistent with expectations as you mentioned! Thank you for flagging!

Reply:
Related to controlled components, the prop types for `value` and `onChange` are not optional, does this mean they should always be used in a controlled way?

Reply:
hmmm yeah, and I think that makes sense! parent components should be the ones managing state for these fields.